crates/sui-node/src/lib.rs

```diff
@@ -917,7 +920,7 @@ pub async fn build_server(
         server.register_module(TransactionExecutionApi::new(
             state.clone(),
             transaction_orchestrator.clone(),
-            state.module_cache.clone(),
+            epoch_store.module_cache().clone(),
```

Comment:
it looks to me like this is going to hold on to whatever module cache existed when the server was created. I'm not sure whether this is going to be easy to fix.

Fortunately TransactionOrch is outside the safety critical core so we don't have to be too paranoid. I suspect that the right thing would be to remove this argument, and instead have TxnOrch call `state.load_epoch_store_one_call_per_task().module_cache()` whenever it needs the module cache.

@longbowlu Thoughts?

Reply:
TxnOrch does not need the module cache. This is instead used in the [rpc translation layers](https://github.com/MystenLabs/sui/blob/main/crates/sui-json-rpc/src/transaction_execution_api.rs#L118) when we return the response back to clients.

But Mark is right that passed reference is not gonna change along with epoch so it's problematic

Reply:
so I removed this call and I am later using the state to fetch the cache. That should give the latest module cache according to the state.
Thoughts?

Reply:
sounds good!